apps/api/src/lib/ranker.ts

```
@@ -113,10 +113,9 @@ async function performRanking(
       return [];
     }
 
-    // Sanitize search query by removing null characters and other control characters
+    // Sanitize search query by removing control characters (\0 is already covered by \x00-\x1F)
     const sanitizedQuery = searchQuery
-      .replace(/\0/g, "") // Remove null characters
-      .replace(/[\x00-\x1F\x7F]/g, "") // Remove control characters
+      .replace(/[\x00-\x1F\x7F]/g, "") // Remove all control characters including null (\0)
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)